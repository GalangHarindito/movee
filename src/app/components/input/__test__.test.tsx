import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";

import SearchInput from "./search";

describe("Search Input Test Component", () => {
  const onchange = jest.fn();
  it("renders search input layout unchanged", () => {
    const { container } = render(
      <SearchInput placeHolder="placeholder" onChange={onchange} />
    );
    expect(container).toMatchSnapshot();
  });

  it("search input should type text", () => {
    const placeholder = 'placeholder'
    const setup = () => {
      const utils = render(
        <SearchInput placeHolder={placeholder} onChange={onchange} />
      );
      const input = screen.getByDisplayValue('');
      const placeholderText = screen.getByPlaceholderText(placeholder)
      return {
        input,
        placeholderText,
        ...utils
      };
    };

    const { input, placeholderText } = setup();
    expect(placeholderText).toHaveAttribute("placeholder", "placeholder")
    fireEvent.change(input, { target: { value: "Die Hard" } });
  });
});
