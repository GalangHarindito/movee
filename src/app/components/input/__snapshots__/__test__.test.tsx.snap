// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Input Test Component renders search input layout unchanged 1`] = `
<div>
  <div
    class="sc-bdfCDU"
  >
    <input
      class="sc-gsTDqH ksnCEC"
      id="input-search"
      placeholder="placeholder"
      type="search"
    />
  </div>
</div>
`;
